### PR TITLE
fix concurrent map writes

### PR DESCRIPTION
**Pull request by xiusin**
concurrent map writes #1296

### CODE REVIEW

File: chromedp.go [hunk 1]
```diff
@@ -351,7 +351,9 @@ func (c *Context) newTarget(ctx context.Context) error {
 			if err != nil {
 				return err
 			}
+			c.Target.frameMu.Lock()
 			c.Target.frames[tree.Frame.ID] = tree.Frame
+			c.Target.frameMu.Unlock()
 			c.Target.cur = tree.Frame.ID
```
Comment (alexandear) [inline chromedp.go:357]: `frameMu` protects both frames and cur. I suppose, should be:
```suggestion
			c.Target.frameMu.Lock()
			c.Target.frames[tree.Frame.ID] = tree.Frame
			c.Target.cur = tree.Frame.ID
			c.Target.frameMu.Unlock()
```

**Review by ZekeLu — COMMENTED** [review]
We should take the changes from #1297 too.

And if you don't mind, please fix other minor issues related `frameMu` listed below:

1. it seems  that a read lock is enough:

https://github.com/chromedp/chromedp/blob/0b936bf02d9c76500cc82f0c809071df2ca97733/chromedp.go#L622-L624

2. `// frameMu protects frames, execContexts, and cur.`
https://github.com/chromedp/chromedp/blob/0b936bf02d9c76500cc82f0c809071df2ca97733/target.go#L31

**Comment by xiusin** [comment]
@ZekeLu This should be left to the friends who have used it deeply before to PR. I give up this change, and I will manually adjust the temporary code locally and compile it for use.

**Comment by ZekeLu** [comment]
Sorry, I meant to drop the review message to #1111. I can not delete the message, my only choice is to hide it. Sorry for the noises.

**Pull request closed without merging**